spec/controllers/categories_controller_spec.rb

```diff
@@ -1,71 +1,30 @@
 RSpec.describe CategoriesController, type: :controller do
-  before do
-    allow(Core::CategoryTreeReader).to receive(:new) do
-      instance_double(Core::CategoryTreeReader, call: double)
-    end
-  end
-
   describe 'GET show' do
-    let(:category) { Core::Category.new('test') }
+    let(:category) { Mas::Cms::Category.new('test') }
     let(:breadcrumbs) { double }
 
     context 'when the category exists' do
       before do
-        allow(Core::CategoryReader).to receive(:new) do
-          instance_double(Core::CategoryReader, call: category)
-        end
+        allow(Mas::Cms::Category).to receive(:find)
+          .with(category.id, locale: I18n.locale)
+          .and_return(category)
 
         allow(BreadcrumbTrail).to receive(:build) { breadcrumbs }
-      end
 
-      it 'is successful' do
         get :show, id: category.id, locale: I18n.locale
+      end
 
+      it 'is successful' do
         expect(response).to be_ok
       end
 
       it 'assigns @category to the result of category reader' do
-        get :show, id: category.id, locale: I18n.locale
-
         expect(assigns(:category)).to eq(category)
       end
 
       it 'assigns @breadcrumbs to the result of breadcrumb reader' do
-        get :show, id: category.id, locale: I18n.locale
-
         expect(assigns(:breadcrumbs)).to eq(breadcrumbs)
       end
     end
-
-    context 'when the category does not exist' do
```

Comment:
Just curious - have these routing error/redirect tests been removed because this stuff is all handled in the ApplicationController now? Do you know if this is being tested elsewhere?

Reply:
yeah these should probably be tested in the application controller

Reply:
This was done in 7e25924